resolver/src/system_conf/unix.rs

```diff
@@ -0,0 +1,406 @@
+        search.push(Name::from_str(&search_domain).map_err(|e| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                format!("Error parsing resolv.conf: {:?}", e),
+            )
+        })?);
+    }
+
+    let config = ResolverConfig::from_parts(domain, search, nameservers);
+
+    let mut options = ResolverOpts::default();
+    options.ndots = parsed_config.ndots as usize;
+    options.timeout = Duration::from_secs(parsed_config.timeout as u64);
+    options.attempts = parsed_config.attempts as usize;
+
+    Ok((config, options))
+}
+
+#[cfg(test)]
+mod tests {
+    use std::env;
+    use std::fs::File;
+    use std::io::Read;
+    use std::net::*;
+    use std::str::FromStr;
+    use std::time::Duration;
+    use trust_dns_proto::rr::Name;
+    use super::*;
+
+    fn empty_config() -> ResolverConfig {
+        ResolverConfig::from_parts(None, vec![], vec![])
+    }
+
+    fn nameserver_config(ip: &str) -> [NameServerConfig; 2] {
+        let addr = SocketAddr::new(IpAddr::from_str(ip).unwrap(), 53);
+        [
+            NameServerConfig {
+                socket_addr: addr,
+                protocol: Protocol::Udp,
+            },
+            NameServerConfig {
+                socket_addr: addr,
+                protocol: Protocol::Tcp,
+            },
+        ]
+    }
+
+    fn tests_dir() -> String {
+        let server_path = env::var("TDNS_SERVER_SRC_ROOT").unwrap_or(".".to_owned());
+        format!{"{}/../resolver/tests", server_path}
+    }
+
+    #[test]
+    fn test_name_server() {
+        let parsed = parse_resolv_conf("nameserver 127.0.0.1").expect("failed");
+        let mut cfg = empty_config();
+        let nameservers = nameserver_config("127.0.0.1");
+        cfg.add_name_server(nameservers[0]);
+        cfg.add_name_server(nameservers[1]);
+        assert_eq!(cfg, parsed.0);
+        assert_eq!(ResolverOpts::default(), parsed.1);
+    }
+
+    #[test]
+    fn test_search() {
+        let parsed = parse_resolv_conf("search localnet.").expect("failed");
+        let mut cfg = empty_config();
+        cfg.add_search(Name::from_str("localnet.").unwrap());
+        assert_eq!(cfg, parsed.0);
+        assert_eq!(ResolverOpts::default(), parsed.1);
+    }
+
+    #[test]
+    fn test_domain() {
+        let parsed = parse_resolv_conf("domain example.com").expect("failed");
+        let mut cfg = empty_config();
+        cfg.set_domain(Name::from_str("example.com").unwrap());
+        assert_eq!(cfg, parsed.0);
+        assert_eq!(ResolverOpts::default(), parsed.1);
+    }
+
+    #[test]
+    fn test_read_resolv_conf() {
+        read_resolv_conf(format!("{}/resolv.conf-simple", tests_dir())).expect("simple failed");
+        read_resolv_conf(format!("{}/resolv.conf-macos", tests_dir())).expect("macos failed");
+        read_resolv_conf(format!("{}/resolv.conf-linux", tests_dir())).expect("linux failed");
+    }
+
+    // #[test]
+    // fn test_ip_addr() {
```

Comment:
The rest of these tests seem to be all commented out. Do you think we should just remove them?

Reply:
I'm open to this.